DatabaseServer/database_server.py

```diff
@@ -339,24 +338,29 @@ def _get_iocs_not_to_stop() -> list:
     SERVER.createPV(BLOCKSERVER_PREFIX, DatabaseServer.generate_pv_info())
     SERVER.createPV(MACROS["$(MYPVPREFIX)"], ExpData.EDPV)
 
-    # Initialise IOC database connection
-    try:
-        ioc_data = IOCData(IocDataSource(SQLAbstraction("iocdb", "iocdb", "$iocdb")), ProcServWrapper(),
-                           MACROS["$(MYPVPREFIX)"])
-        print_and_log("Connected to IOCData database", INFO_MSG, LOG_TARGET)
-    except Exception as e:
-        ioc_data = None
-        print_and_log("Problem initialising IOCData DB connection: {}".format(traceback.format_exc()),
-                      MAJOR_MSG, LOG_TARGET)
-
-    # Initialise experimental database connection
-    try:
-        exp_data = ExpData(MACROS["$(MYPVPREFIX)"], ExpDataSource())
-        print_and_log("Connected to experimental details database", INFO_MSG, LOG_TARGET)
-    except Exception as e:
-        exp_data = None
-        print_and_log("Problem connecting to experimental details database: {}".format(traceback.format_exc()),
-                      MAJOR_MSG, LOG_TARGET)
+
+    ioc_data = None
+    exp_data = None
+
+    while ioc_data is None or exp_data is None:
+        # Initialise IOC database connection
+        if ioc_data is None:
+            try:
+                ioc_data = IOCData(IocDataSource(SQLAbstraction("iocdb", "iocdb", "$iocdb")), ProcServWrapper(), MACROS["$(MYPVPREFIX)"])
+                print_and_log("Connected to IOCData database", INFO_MSG, LOG_TARGET)
+            except Exception:
+                print_and_log("Problem initialising IOCData DB connection: {}".format(traceback.format_exc()), MAJOR_MSG, LOG_TARGET)
+
+        # Initialise experimental database connection
+        if exp_data is None:
+            try:
+                exp_data = ExpData(MACROS["$(MYPVPREFIX)"], ExpDataSource())
+                print_and_log("Connected to experimental details database", INFO_MSG, LOG_TARGET)
+            except Exception:
+                print_and_log("Problem connecting to experimental details database: {}".format(traceback.format_exc()), MAJOR_MSG, LOG_TARGET)
+        
+        # Wait before trying to connect again.
+        sleep(15)
```

Comment:
Won't this cause a 15-second sleep during startup even if both ioc_data and exp_data are immediately available?

How about something like:

```python
if ioc_data is not None and exp_data is not None:
    break
else:
    sleep(15)
```